js.go

```diff
@@ -984,6 +981,12 @@ func (js *js) subscribe(subj, queue string, cb MsgHandler, ch chan *Msg, isSync
 		if err != nil {
 			return nil, err
 		}
+
+		// With flow control enabled async subscriptions we will disable msgs limits,
```

Comment:
Not sure we should disable. We should make sure FC on its on will not trip internal pending limits under normal operations.

Reply:
Updated comment to mention that not disabling but instead increasing the default limits, I did some changes to the way that flow control is done so that it happens after `Ack()` but I can see it still overrun the default pending bytes sometimes.

Reply:
Pending Bytes or Messages?

Reply:
The pending bytes if I use 4K payloads.

Reply:
FC will allow a max of 1M outstanding. So if FC acks are after processing then pending bytes default should be easily sufficient IIRC.